encodings/zigzag/src/compress.rs

```diff
@@ -12,37 +11,35 @@ use crate::ZigZagArray;
 pub fn zigzag_encode(parray: PrimitiveArray) -> VortexResult<ZigZagArray> {
     let validity = parray.validity();
     let encoded = match parray.ptype() {
-        PType::I8 => zigzag_encode_primitive::<i8>(parray.maybe_null_slice(), validity),
-        PType::I16 => zigzag_encode_primitive::<i16>(parray.maybe_null_slice(), validity),
-        PType::I32 => zigzag_encode_primitive::<i32>(parray.maybe_null_slice(), validity),
-        PType::I64 => zigzag_encode_primitive::<i64>(parray.maybe_null_slice(), validity),
+        PType::I8 => zigzag_encode_primitive::<i8>(parray.into_maybe_null_slice(), validity),
+        PType::I16 => zigzag_encode_primitive::<i16>(parray.into_maybe_null_slice(), validity),
+        PType::I32 => zigzag_encode_primitive::<i32>(parray.into_maybe_null_slice(), validity),
+        PType::I64 => zigzag_encode_primitive::<i64>(parray.into_maybe_null_slice(), validity),
         _ => vortex_bail!(
             "ZigZag can only encode signed integers, got {}",
             parray.ptype()
         ),
     };
-    let zz = ZigZagArray::try_new(encoded.into_array())?;
-    zz.inherit_statistics(parray.statistics());
```

Comment:
The compressor does this already and we need a better interface to inherit stats that doesn't require a borrowed value and clone